pyspecProcScripts/phasing.py

```diff
@@ -172,190 +172,150 @@ def costfun(ph1):
-    s.reorder(['t2'],first=False)
-    if fl:
-        fig_forlist, ax_list = plt.subplots(4,1, figsize=(10,10))
-        fl.next("hermitian diagnostic", fig=fig_forlist)
-        fig_forlist.suptitle(" ".join(["Hermitian Diagnostic"] + [j for j in [fl.basename] if j is not None]))
-        fl.image(s.C.mean_all_but(['shift','t2']), ax=ax_list[0])
-        ax_list[0].set_title('Shifted Data')
-    #}}}
-    #{{{make sure there's an odd number of points and set phase of center point to 0
-    logging.debug(strm("before taking the slice, endpoints are",s.getaxis('t2')[r_[0,-1]]))
-    logging.debug(strm("closest to 0", s.getaxis('t2')[np.argmin(abs(s.getaxis('t2')-0))]))
-    # {{{ this passes the assertion, but I
-    #     need jf_hack for the next step to
-    #     work -- this means the axis isn't
-    #     made of numerically exactly equal
-    #     steps, for some reason
-    jf_hack = True
-    test = np.diff(s.getaxis('t2'))
-    assert np.allclose(dt*np.ones_like(test),test)
-    if jf_hack:
-        x = s.getaxis('t2')
-        x[:] /= dt
-        x[:] = np.round(x)
-        x *= dt
-    # }}}
-    s_hermitian = s['t2':(-max_shift,max_shift)].C
-    logging.debug(strm("closest to 0", s.getaxis('t2')[np.argmin(abs(s.getaxis('t2')-0))]))
-    n_points = ndshape(s_hermitian)['t2']
-    logging.debug(strm('size of axis',len(s_hermitian.getaxis('t2'))))
-    logging.debug(strm('steps',np.unique(np.diff(s_hermitian.getaxis('t2')))))
-    logging.debug(strm('full list of axis after slice',s_hermitian.getaxis('t2')))
-    logging.debug(strm("check for endpoints",s_hermitian.getaxis('t2')[r_[0,-1]],
-        s_hermitian.getaxis('t2')[0] + s_hermitian.getaxis('t2')[-1],
-        s_hermitian.getaxis('t2')[n_points//2]))
-    if n_points % 2 == 0:
-        raise ValueError("Getting an even slice -- shouldn't happen if your "
-                "axis coord are in register with t=0 and you take a slice "
-                "(-max_shift,max_shift)")
-    logging.debug("check for center", n_points, s_hermitian.getaxis('t2')[n_points//2:n_points//2+3])
-    assert s_hermitian.getaxis('t2')[n_points//2] == 0
-    center_point = s_hermitian['t2',n_points//2]
-    logging.debug(strm(center_point/abs(center_point)))
-    #I want to do this center_point = s_hermitian['t2':0], but why not be consistent
-    s_hermitian /= center_point/abs(center_point)
-    if fl:
-        fl.image(s_hermitian.C.mean_all_but(['shift','t2']),
-                ax=ax_list[1])
-        ax_list[1].set_title('Hermitian Data')
+    orig_dt = s.get_ft_prop(direct, "dt")
+    s.ft(direct)
+    s = s[direct:frq_range] 
+    s.ift(direct,pad=2048*8)
+    if fl is not None:
+        fig, ax_list = subplots(6, 1, figsize=(10,10))
+        fl.next('Hermitian Function Test Diagnostics',fig=fig)
+        s = s[direct:(ini_delay,None)]
+        fl.plot(abs(s),'.',ax=ax_list[0])
+        ax_list[0].set_title('Data with Padding')
```

Comment:
this set of figures is very hard to see as-is when I run it from the command line.  Can you break it up into a couple different figures, so that we can actually see the subplots?  Or maybe make it into a 2 column grid that's wider in size?

Reply:
this is not fixed (I checked the "outdated" here, and it's just b/c I ran black on the code)

Reply:
(I did make sure I pulled and was on the right branch this time!!!)

Reply:
oops sorry! okay hopefully now it's good I changed it to a 2 x 3 plot